orttraining/orttraining/training_ops/cuda/tensor/split.cc

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "split.h"
+#include "core/providers/cuda/tensor/split_impl.h"
+#include "core/providers/cpu/tensor/utils.h"
+#include "core/providers/common.h"
+
+namespace onnxruntime {
+namespace cuda {
+ONNX_OPERATOR_KERNEL_EX(SplitTraining,
+                        kMSDomain,
+                        1,
+                        kCudaExecutionProvider,
+                        KernelDefBuilder()
+                            .InputMemoryType<OrtMemTypeCPUInput>(1)
+                            .TypeConstraint("T", DataTypeImpl::AllFixedSizeTensorTypes()),
+                        SplitTraining);
+
+Status SplitTraining::PrepareForCompute(const TensorShape& input_shape, int num_outputs, int64_t& axis, int& before_dims,
```

Comment:
can be reused from cpu implementation by importing orttraining/orttraining/training_ops/cpu/tensor/split.h

Reply:
not straightforward to reuse, might have to pull PrepareForCompute out of class, like ReduceSum does.
will try that.

---
In reply to: [459771005](https://github.com/microsoft/onnxruntime/pull/4597#discussion_r459771005) [](ancestors = 459771005)

Reply:
do-able - need to make sure namespace match

---
In reply to: [459806870](https://github.com/microsoft/onnxruntime/pull/4597#discussion_r459806870) [](ancestors = 459806870,459771005)